webgpu-compute-demo/script.js

```diff
@@ -0,0 +1,133 @@
+// Define global buffer size
+const BUFFER_SIZE = 1000; 
+
+// Compute shader
+const shader = `
+@group(0) @binding(0)
+var<storage, read_write> output: array<f32>;
+
+@compute @workgroup_size(64)
+fn main(
+  @builtin(global_invocation_id)
+  global_id : vec3u,
+
+  @builtin(local_invocation_id)
+  local_id : vec3u,
+) {
+  // Avoid accessing the buffer out of bounds
+  if (global_id.x >= ${BUFFER_SIZE}) {
+    return;
+  }
+
+  output[global_id.x] =
+    f32(global_id.x) * 1000. + f32(local_id.x);
+}
+`;
+
+// Main function
+
+async function init() {
+  // 1: request adapter and device
+  if (!navigator.gpu) {
+    throw Error('WebGPU not supported.');
+  }
+
+  const adapter = await navigator.gpu.requestAdapter();
+  if (!adapter) {
+    throw Error('Couldn\'t request WebGPU adapter.');
+  }
+
+  const device = await adapter.requestDevice();
```

Comment:
**🎯 suggestion:** I think it might be helpful to rename `device` to `gpuDevice ` for clarity.